### PR TITLE
Fix couple flaky tests

### PR DESCRIPTION
**Pull request by BrennanConroy**

### CODE REVIEW

File: test/Microsoft.AspNetCore.SignalR.Client.Tests/HubConnectionTests.cs [hunk 1]
```diff
@@ -98,6 +98,7 @@ public class HubConnectionTests
 
             await hubConnection.StartAsync();
             await hubConnection.DisposeAsync();
+            await hubConnection.Closed.OrTimeout();
```
Comment (moozzyk) [inline test/Microsoft.AspNetCore.SignalR.Client.Tests/HubConnectionTests.cs]: this seems wrong - hubConnection has been disposed so the expectation that the `Closed` task is valid is not correct. The bug is somewhere else...

Comment (BrennanConroy) [reply test/Microsoft.AspNetCore.SignalR.Client.Tests/HubConnectionTests.cs]: Yeah I wasn't sure about that. We could await the closed task in DisposeAsync?

Comment (moozzyk) [reply test/Microsoft.AspNetCore.SignalR.Client.Tests/HubConnectionTests.cs]: Yes, it has to be something to this effect

File: src/Microsoft.AspNetCore.SignalR.Client.Core/HubConnection.cs [hunk 2]
```diff
@@ -334,7 +334,7 @@ private Task InvokeStreamCore(string methodName, InvocationRequest irq, object[]
             }
         }
 
-        private Task Shutdown(Exception ex = null)
+        private void Shutdown(Exception ex = null)
```
Comment (moozzyk) [inline src/Microsoft.AspNetCore.SignalR.Client.Core/HubConnection.cs]: :+1:

**Pull request merged**